packages/jest-cli/src/TestRunner.js

```diff
@@ -273,14 +275,22 @@ class TestRunner {
     return Promise.race([runAllTests, onInterrupt]).then(cleanup, cleanup);
   }
 
+  _shouldAddDefaultReporters(reporters?: Array<ReporterConfig>): boolean {
+    return !reporters || reporters.indexOf(DEFAULT_REPORTER_LABEL) !== -1;
+  }
+
   _setupReporters() {
     const config = this._config;
+    const reporters = config.reporters;
+    const isDefault: boolean = this._shouldAddDefaultReporters(reporters);
```

Comment:
does this really need a type annotation inline? Flow should be able to figure it out.